Src/PCLMock/WhenContinuation.cs

```diff
@@ -346,15 +347,26 @@ internal override object Apply(object mockedObject, params object[] args)
         {
             Debug.Assert(args != null);
 
-            if (this.exception != null)
+            if (this.doAction != null)
```

Comment:
Checking the `doAction` first now so that it executes even if there's a throw afterwards.